### DISCUSSION TITLE
I built Backlinking for Docusaurus!

### DISCUSSION THREAD

**Discussion by anaclumos** [facebook/docusaurus#8217] [Show and tell]
Reference: #7610

I built a backlinking system for docusaurus!

Repo: [anaclumos/extracranial](https://github.com/anaclumos/extracranial)

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/31657298/196146684-9f262a9e-8766-408f-b595-bee19e2f5c03.png">

**Comment by yosevu** [comment]
Nice @anaclumos! I was just doing research to see if anyone had done this. Thanks for open-sourcing your work! I'll take a look. I also started looking at [Docusaurus plugins](https://docusaurus.io/docs/advanced/plugins) to see if this kind of functionality can be generalized as a plugin.

**Reply by anaclumos** [reply]
Thanks! I want Docusaurus to have this feature built-in 😃

Just to explain how it works, a [python preprocessor](https://github.com/anaclumos/extracranial/blob/main/tools/process-backlinks.py) will parse the document before building the site. This is necessary because I use `[[wikilink]]` for interlinking, which Docusaurus does not support. I have another python script that converts wikilinks to docusaurus-recognizable markdown links. After that, the [corresponding component](https://github.com/anaclumos/extracranial/blob/main/src/components/BacklinkTable/index.tsx) will simply read the indexed backlink JSON to display it on the client side.

**Reply by slorber** [reply]
That looks interesting.
Technically we support remark plugins and you could implement support for wikilinks syntax with a remark plugin, that would be better integrated into your build. Now we probably don't have an easy way to implement such preprocessor in Docusaurus (yet)

**Reply by yosevu** [reply]
@slorber awesome that Docusaurus supports remark plugins!

I [created a plugin](https://github.com/yosevu/docusaurus-smart-notes/blob/main/src/plugins/smart-notes/transformWikilinks.js) to test transforming Wikilinks using https://github.com/landakram/remark-wiki-link.

I'm currently trying to understand the best way to use the Docusaurus plugin system and Lifecycle hooks to collect Wikilinks for each markdown blog post in order to render in the `BlogSidebar`. [Here is what I have so far](https://github.com/yosevu/docusaurus-smart-notes/blob/main/src/plugins/smart-notes/collectBacklinks.js).

I think I have to use the `createData` hook, but I don't see how to pass the data to an existing blog post route as `BlogPostPage` props. Is this possible?

**Reply by slorber** [reply]
I'm not 100% sure to understand what you want to do exactly, but we probably don't have good enough apis to do what you want to do yet unfortunately.

**Comment by Shake-Jarjane** [comment]
Hello, I am very interested in this backlink, but I do not know about this aspect, can you teach me the detailed installation tutorial? Thanks!

**Reply by anaclumos** [reply]
Hi, thanks for commenting. I may pack the docusaurus platform into a prepackaged second-brain framework (@stevejkang has inspiring thoughts on this) in the future, but currently, I am busy and cannot do it. However, the details you are looking for are open-sourced at https://github.com/anaclumos/extracranial, so please take a look. Thank you.

**Reply by Shake-Jarjane** [reply]
Okay, thank you for your reply
